### ISSUE TITLE
Cloak of Shadow bug.

### ISSUE THREAD

**Issue by Kinglundh** [ccshiro/cc-buglist#530]
When a mage is casting a frostbolt, and its "in the air", and the Rogue is using cloak, the frostbolt still gets through. Seems like Cloak of Shadows aint working on spells that got casted and is close to your character.

**Comment by Ickler** [comment]
This is working as intended, once a spell is over halfway there you can't avoid it with Cloak of Shadows (afaik). Same goes for Vanish, if you vanish past the midway point of the traveltime you will be taken out of vanish, although take no damage.

**Comment by Kinglundh** [comment]
You might have a point there Ickler. :-)

**Comment by Klimpb** [comment]
Cloak of shadow is not like a bubble, it will have 0 effect on any spell that was casted before cloak of shadows was applied.

**Comment by Ickler** [comment]
It does, as long as it was used before the spell has traveled beyond the halfway point.

**Comment by Soda-88** [comment]
@Ickle you're wrong, SuddenDeath is correct
It works exactly the same as Warrior's Spell Reflection, if you use it even a microsecond after the spell has been launched, it will go through

There has never been any 'halfway points' in history of WoW, it was made up by CC developers for lack of better judgement as shown in their early 'Rogue fixes' video which I disputed with series of clips from Neilyo's videos.

**Comment by Klimpb** [comment]
@Ickler if I remember correctly, cloak of shadows was changed in cataclysm to work as you're describing it.

**Comment by Ickler** [comment]
Not sure how it's implemented on CC right now, but I think it's the way I said. And I still believe it's supposed to be that way. But either way, to change it one would need to do some research and provide proof.

**Comment by Soda-88** [comment]
@Ickler stop spreading disinformation please. Cloak has been identical to Spell Reflection up until Cataclysm when they swapped Vanish and Cloak functionality, i.e. prior to Cataclysm you used Vanish to dodge airborne spells, and post Cataclysm you used CloS to dodge airborne spells. The way you describe it (without the halfway travel mumbo-jumbo) is the way it works of Feenix and it's just wrong.

**Comment by Ickler** [comment]
@Soda-88 Well as for halfway points or not, how do you explain the situation with vanish then? Because it was clearly something on retail that made you avoid damage but lose stealth if you were "slow" on the vanish. However if you vanish while it's just leaving, you take no damage, and keep stealth. 

Now since it's several years since retail I can't recall 100%, I could be wrong. But I'd still say you have the burden of proof. 
It's not spreading disinformation. Comments here, are just that comments :) Devs take it for what it is, my memory is not fact. 

And if in fact it is implemented the way I described it on CC, you have to test it next test, and report it if you believe it to be wrong. And provide sufficient proof to back up your claims.

**Comment by Soda-88** [comment]
@Ickler it wasn't halfway point, it was specifically if you vanished at the same moment you got hit by a spell, hence why you NEVER ended up in stealth if you vanished instant cast spells that have no projectiles such as Blind, Fire Blast, etc. But even then you'd avoid the spell effect, it just broke the stealth instantly. Here's a video compilation of all the clips: https://mega.nz/#!o4hAlCTa!uZjG8c4kYp0MojhqlcwRlT7tHQOyklu6JLAAzeNViWQ

**Comment by Ickler** [comment]
@Soda-88 Stop using this as a discussion topic please. And as I said if you believe it to be wrong, find your proof and open a specific issue about the issue after testing it on next test.

Edit: That video has nothing to do with cloak of shadows, I struggle to see why you even linked it. I can think of a couple of movies where you can clearly see a cloak being used, right after a spell, yet it still gets resisted. I'll look through some old vid's and find some stuff for you, should you ever report it as an issue :) 
Also claiming that it is using the same mechanics as spell reflect requires some more detail, not just your bold statement (even though from what I know, and my non-existing programming skills, it might be possible to tell from the code itself).

**Comment by Soda-88** [comment]
@Ickler face meets palm
How about you take your own advice and provide the video that this in fact is a bug instead? I clearly stated that this is not a bug. And as for me providing a proof, I doubt there is a single PvP video of a rogue misusing Cloak out there the way you described (cloaking after the projectile has been launched) but if you can find one, go ahead :)
And if the video I posted in my previous post doesn't reinforce the fact that only Vanish was used to avoid airborne projectiles, I don't know what will.

P.S. I sincerely apologize to the dev team

**Comment by Ickler** [comment]
@Soda-88 I told myself I wouldn't reply to this more, but your train of thought and logic just has to be pointed out to you... You're arguing about Cloak of Shadows not being able to resist a spell already airborne when cloak is used. And your proof of this is posting a video of a few clips showing that Vanish is used to avoid spells in mid-air and instant spells... 
Then you say this: 
"And if the video I posted in my previous post doesn't reinforce the fact that only Vanish was used to avoid airborne projectiles, I don't know what will." .....

With that logic that means the video you posted show ALL instances that ever happened where a spell was avoided, and in all those instances Vanish was the only spell used! If you can't even see how folly your arguments are then I can't take anything you say or write seriously. Nor should anyone else... 

As I said, I might be wrong.. I 'm willing to accept that :) It doesn't seem like you are though. And now let's end it, and pm me on IRC if you want to keep nagging.

**Comment by Soda-88** [comment]
@Ickler My logic is that the most popular rogue player in TBC during retail era never thought of using Cloak of Shadows to avoid airborne spells BECAUSE IT NEVER WORKED THAT WAY (until Cataclysm, hence why he always used Vanish). I'm trying to dispute your 'halfway point' argument here, if spell launched, and no matter how far it traveled, it will ignore Cloak of Shadows that has been used AFTER the spell was launched, no matter if projectile is still in first or second half of its travelling trajectory.

To sum it up: OP is wrong, Ickler is semi-right (there is no halfway points), SuddenDeath is right

**Comment by vexxedd** [comment]
Well. There is no discussion about it. Funny to see coments like "I think". There is no "I think". You know or you dont, you've played tbc retail or you did not.
If you use Cloak when spell is already launched - it will go through. Basically, like people said before its working like Spell reflection. 
Vanish on the other hand works different. There is no "half-way". If you didnt get affected by the spell you can vanish any time to avoid it.

**Comment by somnu** [comment]
I played rogue for over 3 years in TBC and WOTLK, I was a neilyo fanboy, I always liked trying new tricks and abusing the class, I ended up knowing the rogue class inside out mechanic-wise. And I can say with complete certainty that Cloak of Shadows did not cause you to resist the spell if the spell is already airborne, this also applied for Spell Reflection.

**Comment by ccshiro** [comment]
As others in the thread have already said; like spell reflection the result is decided when the spell leaves your hands, not when it hits. Basically, working as intended.
